### PR TITLE
libsdl2: Remove BDirectWindow, fix context management

### PR DESCRIPTION
**Pull request by X547**
WIP, do not merge.

### CODE REVIEW

**Comment by pulkomandy** [comment]
SDL2 maintainers wuld prefer that such large patches are submitted to them as well. The Haiku port is upstreamed and it's simpler for everyone if they can get the patches there directly.

File: media-libs/libsdl2/patches/libsdl2-2.0.10.patchset [hunk 1]
```diff
@@ -1292,3 +1292,581 @@ index 8d8da32..d8950c2 100644
...
+         _draw_thread_id = spawn_thread(HAIKU_DrawThread, "drawing_thread",
+                             B_NORMAL_PRIORITY, (void*) this);
+         resume_thread(_draw_thread_id);
+ #endif
++    	printf("+SDL_BWin}\n");
+     }
+ 
+     virtual ~ SDL_BWin()
+     {
++    	printf("-SDL_BWin{\n");
```
Comment (korli) [inline media-libs/libsdl2/patches/libsdl2-2.0.10.patchset]: leftover?

**Comment by waddlesplash** [comment]
We already have 1.2k lines of patches that nobody upstreamed :/

**Comment by pulkomandy** [comment]
They were upstreamed: https://bugzilla.libsdl.org/show_bug.cgi?id=4442
We're just waiting on a new SDL release to remove the patch here. So please do not add to the patch here and work with upstream, thanks.

**Comment by X547** [comment]
Uploaded version ready to merge.

**Comment by X547** [comment]
Can someone suggest SDL2 applications to test? I already tested Blender, VVVVVV and some simple testing applications.

**Comment by threedeyes** [comment]
> Can someone suggest SDL2 applications to test? I already tested Blender, VVVVVV and some simple testing applications.

For example:
PPSSPP - PSP emulator (required PSP ROMs)
OpenMW - Morrowind game (required game resources from original morrowind)
Arx libertatis - Arx Fatalis game (required game resources from original game)
Astomenace - Scroller arcade game
Colobot -  Real-time strategy game

**Comment by kenmays** [comment]
@X547  - See: https://bugzilla.libsdl.org/show_bug.cgi?id=4442
Needs response for SDL 2.0.14 merge.

**Comment by waddlesplash** [comment]
What else is needed here, and can it be submitted upstream?

**Comment by waddlesplash** [comment]
@X547 you said in January that this was ready to be merged, is that still true? Or did you find problems?

**Comment by waddlesplash** [comment]
Ah, never mind, I see there are hacks in the code re. `BGL_SHARE_CONTEXT`, and also @pulkomandy's comment about upstreaming. So we should upstream Mesa shared contexts first I suppose.

**Comment by diversys** [comment]
If this fixes Blender preferences (#4466) maybe we can still merge this and worry about upstreaming later?

**Comment by pulkomandy** [comment]
Good news for you, SDL has recently moved to github: https://github.com/libsdl-org/SDL so, no separate account is needed.

**Comment by waddlesplash** [comment]
Any chance this can be submitted upstream to SDL2 now that we are on the newer Mesa?

**Comment by threedeyes** [comment]
> Any chance this can be submitted upstream to SDL2 now that we are on the newer Mesa?

I have adapted this patch for the SDL 2.0.14 and have been using it for a very long time.
A test build with this patch is available in my repository (http://haikuware.ru/package_list.php?repo=lote).
This patch solves problems with Blender, but for example the game 0 A.D. does not work with it (crash on start).

**Comment by kenmays** [comment]
@X547 - This seems resolved per current libSDL2 2.0.14-7 testing.
- [x ] Haiku's libsdl2 2.0.x legacy patches were approved and integrated upstream within the recent libsdl2 v2.0.15+ releases.
- [x] Blender - visuals and crash on quit were tested and resolved for blender 2.93.5 and libsdl2 2.0.14-7 on Haiku hrev55507 x64. 
- [x] 0 A.D - starts/play/exit game (PASSED). No major issues there. (0.0.23b-a1, Build: Oct 8,2021, 21946P-release)
- [x] julius -  starts/play/exit game (PASSED). No major issues there. (1.6.0-1)
- [x] colobot - starts/play/exit game (PASSED). No major issues there. (0.2.0-1)
- [x ] astromenace - starts/play/exit game (PASSED). No major issues there. (1.4.1-1)

**Comment by kenmays** [comment]
libsdl2 2.0.18-1 is now in repo.

**Pull request closed without merging**